adminSiteServer/apiRouter.ts

```diff
@@ -457,20 +503,27 @@ const saveGrapher = async (
 
     // Execute the actual database update or creation
     let chartId: number
+    let patchConfig: GrapherInterface
+    let fullConfig: GrapherInterface
     if (existingConfig) {
         chartId = existingConfig.id!
-        newConfig = await updateExistingChart(knex, {
+        const configs = await updateExistingChart(knex, {
             config: newConfig,
             user,
             chartId,
         })
+        patchConfig = configs.patchConfig
+        fullConfig = configs.fullConfig
     } else {
-        newConfig = await saveNewChart(knex, {
+        const configs = await saveNewChart(knex, {
             config: newConfig,
             user,
         })
-        chartId = newConfig.id!
+        patchConfig = configs.patchConfig
+        fullConfig = configs.fullConfig
+        chartId = fullConfig.id!
     }
+    newConfig = patchConfig
```

Comment:
Given this, can we always be sure that the fields accessed below exist on the `newConfig` obj? E.g. `isPublished`, `dimensions`, `slug`, etc.?
I believe there can also be cases where some of these are fully inherited, leading to issues.

Reply:
Good point - an earlier version of Sophia's work used the full config here but she changed it since. I think in the medium term we want to pull out slug and isPublished from the config and move it to the charts table but that is a bigger change that we wanted to avoid now. Both of these don't make sense to inherit really and are, for now, set at the final, standalone chart level. For those two, therefore, it doesn't make too much of a difference in practice - but the access of the dimensions property should be done with the full config. 

I'll switch the access around there and only use the patchConfig for storing it in the revision log.